src/System.Linq.Expressions/src/System/Linq/Expressions/Compiler/ILGen.cs

```diff
@@ -947,6 +947,10 @@ private static void EmitNumericConversion(this ILGenerator il, Type typeFrom, Ty
                                 il.Emit(OpCodes.Conv_I8);
                             }
                             break;
+                        case TypeCode.Boolean:
+                            il.Emit(OpCodes.Ldc_I4_0);
```

Comment:
I am just wondering - if C# supported int-->bool cast, how that would work.  

Considering that one could then write:
```cs
if ( (bool)intVar )
{
   . . .
}
``` != 0  ```   does seem to be the right choice